.github/copilot-instructions.md

```diff
@@ -0,0 +1,192 @@
+# tlang - toy language to learn parsing / simple codegen
+
+tlang is a functional programming language implemented in Rust that compiles to JavaScript. It features pattern matching, tail call optimization, algebraic data types (enums), and WebAssembly bindings for browser usage.
+
+Always reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+## Working Effectively
+
+### Bootstrap and Build (NEVER CANCEL - Build times are long)
+**CRITICAL**: All build commands can take 30+ minutes. NEVER CANCEL builds or tests. Use timeouts of 60+ minutes for build commands and 30+ minutes for test commands.
+
+1. **Install Rust toolchain and dependencies:**
+   ```bash
+   # Rust nightly is required (set by rust-toolchain.toml)
+   rustup target add wasm32-unknown-unknown
+   cargo install cargo-nextest        # ~3m 10s installation time
+   cargo install wasm-pack           # ~1m 30s installation time
+   
+   # Install exact wasm-bindgen-cli version (CRITICAL - version must match Cargo.toml)
+   WASM_BINDGEN_VERSION=$(grep -oP 'wasm-bindgen = "=\K[^"]+' crates/tlang_bindings_js/Cargo.toml)
+   cargo install wasm-bindgen-cli --version $WASM_BINDGEN_VERSION  # ~1m 30s installation time
+   ```
+
+2. **Install Node.js dependencies:**
+   ```bash
+   npm ci  # ~12s installation time
+   ```
+
+3. **Download binaryen (required for WebAssembly optimization):**
+   ```bash
+   wget https://github.com/WebAssembly/binaryen/releases/download/version_119/binaryen-version_119-x86_64-linux.tar.gz
+   tar xvzf binaryen-version_119-x86_64-linux.tar.gz
+   export PATH=$(pwd)/binaryen-version_119/bin:$PATH
+   ```
+
+### Build Commands (NEVER CANCEL - Set timeout to 60+ minutes)
+- **Rust tests**: `cargo nextest run --profile=ci` -- takes ~55s. NEVER CANCEL. Set timeout to 30+ minutes.
+- **Full test suite**: `make test` -- takes ~64s. NEVER CANCEL. Set timeout to 30+ minutes.
```

Comment:
We should mention, that the `make test` command also uses Node.js and it should use the node version mentioned in package.json (under the volta key).